Added Demoblaze website UI automation.

annahanoshenko opened the pull request.

Commit:
Added Demoblaze website UI automation.

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Pages/LoginPage.cs:
```diff
@@ -0,0 +1,42 @@
+           EnterUserName(user.UserName);
+           EnterPassword(user.Password);
+           ClickLoginPopUpButton();
+
```
please remove empty space

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Pages/CartPage.cs:
```diff
@@ -0,0 +1,30 @@
+        public void ClickCartButton() => CartButton.Click();
+        public void ClickDeleteProductBtn() => DeleteFromCartBtn.Click();
+        public void ClickPlaceOrderButton() => PlaceOrderButton.Click();
+
```
please remove empty space

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Tests/Tests.cs:
```diff
@@ -0,0 +1,205 @@
+            Assert.That(actualAlertText.Equals(expectedAlertText));
+
+            cartPage.ClickCartButton();
+        }
```
please add additional space

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/DragAndDropGlobalsqaTests/Base/PageBase.cs:
```diff
@@ -0,0 +1,28 @@
+﻿using OpenQA.Selenium.Support.UI;
+using OpenQA.Selenium;
+using SeleniumExtras.WaitHelpers;
+
```
please remove extra space

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/DragAndDropGlobalsqaTests/Tests/DragAndDropTests.cs:
```diff
@@ -0,0 +1,118 @@
+            mainPage.SwitchToPhotoManagerFrame();
+            mainPage.ClickTrashIcon();
+
+            bool isElementInTrash = mainPage.IsElementInTrash(photoName);
```
There is no verification in this test. Just isElementInTrash, but no checking of the result value

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/DragAndDropGlobalsqaTests/Tests/DragAndDropTests.cs:
```diff
@@ -0,0 +1,118 @@
+            mainPage.ClickTrashIcon();
+            mainPage.DragAndDropElement(photoName);
+
+            bool isElementInPhotoMAnager = mainPage.IsElementInPhotoMAnager(photoName);
```
same here

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/DragAndDropGlobalsqaTests/Tests/DragAndDropTests.cs:
```diff
@@ -0,0 +1,118 @@
+            mainPage.ClickTrashIcon();
+            mainPage.ClickRecycleIcon();
+
+            bool isElementInPhotoMAnager = mainPage.IsElementInPhotoMAnager(photoName);
```
same here

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/DragAndDropGlobalsqaTests/Tests/DragAndDropTests.cs:
```diff
@@ -0,0 +1,118 @@
+            mainPage.ClickZoomIcon();
+            mainPage.ClickExitIcon();
+
+            bool isElementInPhotoMAnager = mainPage.IsElementInPhotoMAnager(photoName);
```
same here

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/DragAndDropGlobalsqaTests/Base/ImageWidget.cs:
```diff
@@ -0,0 +1,20 @@
+﻿
```
remove empty space

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Tests/Tests.cs:
```diff
@@ -0,0 +1,205 @@
+        [Test]
+        public void ShouldCteateNewUser_WhenDataIsValid()
+        {
+            UserEntity user = new UserEntity("Anna" + DateTime.Now.Ticks, "qwerty67");
```
Please create class called UserStorage and add all of the users you use in the tests to the storage and then reuse here. 

You can do it in a form of the static property

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Entities/UserEntity.cs:
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
```
Are you sure that you are using all usings in the class? Please check

annahanoshenko replied:
I removed extra usings

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Entities/OrderEntity.cs:
```diff
@@ -0,0 +1,24 @@
+            Month = month;
+            Year = year;
+        }
+
```
Please remove extra space

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Entities/OrderEntity.cs:
```diff
@@ -0,0 +1,24 @@
+﻿
```
same here

annahanoshenko replied:
Done

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblaze/AutoTestsDemoblaze.cs:
```diff
@@ -0,0 +1,220 @@
+namespace DemoblazeUiTAF.AutoTestsDemoblaze
+{
+    [TestFixture]
+    internal class AutoTestsDemoblaze
```
Do you still need these tests? I see that you re-write this tests using POM

If you don't need them - please remove

annahanoshenko replied:
Done!

vladyslavhanoshenko commented:
I see a lot of extra spaces and not used usings. Please clean up them all

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Base/BaseTest.cs:
```diff
@@ -0,0 +1,26 @@
+        public void Setup()
+        {
+            Driver = new ChromeDriver();
+            Driver.Navigate().GoToUrl("https://www.demoblaze.com");
```
Could we please create class called EnvironmentConfig and put website URL there?

annahanoshenko replied:
Done!

Commit:
created class called UserStorage and add all of the users in the tests to the storage

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Tests/Tests.cs:
```diff
@@ -0,0 +1,188 @@
+namespace DemoblazeUiTAF.AutoTestsDemoblazePOM.Tests
+{
+    [TestFixture]
+    public class Tests : BaseTest
```
I see that in this test class we have a lot of tests that can be grouped by feature category. 

Let's create different tests classes based on the feature we are testing, for example:

- RegistrationTests
- LoginTests
- OrderTests
etc...

annahanoshenko replied:
Done!

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Tests/Tests.cs:
```diff
@@ -0,0 +1,188 @@
+        [Test]
+        public void MakeOrderShouldFail_WhenOrderNameAndCardFieldsAreEmpty()
+        {
+            OrderEntity order = new OrderEntity("", "USA", "New York", "", "12", "122029");
```
Could you please create a builder class for the OrderEntity?

1) Let's create builder for OrderEntity
2) Then let's create OrdersStorage and use builder there to create OrderObjects

annahanoshenko replied:
Done

Commit:
Added checking of the result value in tests

vladyslavhanoshenko commented on DemoblazeUiTAF/AutoTestsDemoblazePOM/Base/BaseTest.cs:
```diff
@@ -0,0 +1,26 @@
+        [SetUp]
+        public void Setup()
+        {
+            Driver = new ChromeDriver();
```
I like the idea of having Driver initialization here in BeforeScenario, but I would like to have Singleton pattern here to ensure we have only one point of access for our WebDriver

annahanoshenko replied:
Done

annahanoshenko replied:
Do I need to use Singleton pattern in the "DragAndDropGlobalsqa" project?

vladyslavhanoshenko commented on DemoblazeUiTAF/DragAndDropGlobalsqaTests/Base/BaseTest.cs:
```diff
@@ -0,0 +1,26 @@
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium;
+
+namespace DemoblazeUiTAF.DragAndDropGlobalsqaTests.Base
```
Why do we need two identical BaseTest? Can we reuse another one? 

Let's create folder called Common, move first BaseTest there and reuse it for both

annahanoshenko replied:
It is impossible to create two instances of BaseTest because each instance would contain different website URLs.

vladyslavhanoshenko commented on DemoblazeUiTAF/DragAndDropGlobalsqaTests/Tests/DragAndDropTests.cs:
```diff
@@ -0,0 +1,114 @@
+            mainPage.ClickZoomIcon();
+            mainPage.ClickExitIcon();
+
+            bool isElementInPhotoMAnager = mainPage.IsElementInPhotoMAnager(photoName);
```
I see common capital A after M. Let's make it consistent and call variable isElementInPhotoManager

annahanoshenko replied:
Done

Commit:
Created class called EnvironmentConfig and put website URL there

Commit:
Created different tests classes based on the feature we are testing:
RegistrationTests
LoginTests
OrderTests

Commit:
Created a builder class for the OrderEntity.
Then created OrdersStorage and use builder there to create OrderObjects

Commit:
Used Singleton pattern here to ensure we have only one point of access for our WebDriver

Commit:
1)  Created different tests classes in the "DragAndDropGlobalsqa":
DeleteImageTests
RecycleImageTests
ZoomImageTests

2)made consistent and call variable isElementInPhotoManager
